src/voidZeroRule.ts

```diff
@@ -0,0 +1,44 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+import * as tsutils from 'tsutils';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+
+const FAILURE_STRING: string = 'Replace void 0 with undefined';
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: ExtendedMetadata = {
+        ruleName: 'void-zero',
+        type: 'maintainability',
+        description: 'Avoid using void 0; use undefined instead.',
+        rationale:
+            'void 0, which resolves to undefined, can be confusing to newcomers.\
```

Comment:
Nit: this will produce string with 12 spaces between `.` and `Exclusively`.
<pre>
void 0, which resolves to undefined, can be confusing to newcomers.            Exclusively use undefined to reduce ambiguity.</pre>

Better to just make it single line. Without extra spaces it should fit line limit.

Reply:
Sure!